pkg/publish/default.go

```diff
@@ -204,6 +204,12 @@ func (d *defalt) Publish(ctx context.Context, br build.Result, s string) (name.R
 	// https://github.com/google/go-containerregistry/issues/212
 	s = strings.ToLower(s)
 
+	// safeguarding against the case where the user has specified an empty tag with publish.WithTags([]string{""}) or publish.WithTags([]string{})
+	if len(d.tags) == 0 || (len(d.tags) == 1 && d.tags[0] == "") {
+		log.Printf("No tags specified, using default tag: %s", latestTag)
+		d.tags = []string{latestTag}
+	}
```

Comment:
What do we do today if someone manages to pass empty tags? It might be expected that we publish images by digest and don't tag anything.

If we don't do that today then I'm fine changing that so that callers (presumably via the Go package, and not the CLI) default to using the `:latest` tag. If there's any way to call the CLI so that things end up published but untagged, we should probably retain that behavior. And document it.